pkg/auth/creds_config.go

```diff
@@ -0,0 +1,136 @@
+package auth
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/notaryproject/notation/pkg/config"
+	dockerConfig "github.com/notaryproject/notation/pkg/dockerconfig"
+)
+
+type ConfigFile struct {
+	CredentialsStore  string            `json:"credsStore,omitempty"`
+	CredentialHelpers map[string]string `json:"credHelpers,omitempty"`
+}
+
+// loadConfig loads the configuration
+// from the config file
+func loadConfig() (*ConfigFile, error) {
+	config, _ := loadNotationConfig()
+	if config != nil && !config.isEmptyConfig() {
+		fmt.Println("Using notation config file")
+		return config, nil
+	}
+
+	config, _ = loadDockerConfig()
+	if config != nil && !config.isEmptyConfig() {
+		fmt.Println("Using docker config file")
+		return config, nil
+	}
+	if !config.ContainsAuth() {
+		fmt.Println("Credentials store config is not set up, using the default setting")
+		config.CredentialsStore = DetectDefaultStore(config.CredentialsStore)
+	}
+	return config, nil
+}
+
+// loadDockerConfig loads the configuration
+// from the config file under .notation directory
+func loadNotationConfig() (*ConfigFile, error) {
+	notationConfig, err := config.LoadOrDefault()
+	if err != nil {
+		return nil, err
+	}
+	config := NewConfigFromNotationConfig(notationConfig)
+	return config, nil
+}
+
+// loadDockerConfig loads the configuration
+// from the config file under .docker directory
+func loadDockerConfig() (*ConfigFile, error) {
+	dockerConfig, err := dockerConfig.Load(dockerConfig.Dir())
+	if err != nil {
+		return nil, err
+	}
+	config := NewConfigFromDockerConfig(dockerConfig)
+	return config, nil
+}
+
+// isEmptyConfig checks if the config is empty
+func (configFile *ConfigFile) isEmptyConfig() bool {
+	return configFile.CredentialsStore == "" && len(configFile.CredentialHelpers) == 0
+}
+
+// GetCredentialsStore returns a new credentials store from the settings in the
+// configuration file
+func (configFile *ConfigFile) GetCredentialsStore(registryHostname string) CredentialStore {
+	if helper := getConfiguredCredentialStore(configFile, registryHostname); helper != "" {
+		return newNativeStore(helper)
+	}
+	return nil
+}
+
+// var for unit testing.
+var newNativeStore = func(helperSuffix string) CredentialStore {
+	return NewNativeAuthStore(helperSuffix)
+}
```

Comment:
nit:
```suggestion
// var for unit testing.
var newNativeStore = NewNativeAuthStore
```

Reply:
Good catch! Forgot to update it when I moved.